core/config/options.go

```diff
@@ -307,17 +306,25 @@ func (opts *Options) Set(key string, tup []string) sinkdb.Op {
 	})
 }
 
-// Remove removes the provided tuple from the configuration option set
-// indicated by key.
+// Remove removes the provided tuple from the configuration option
+// indicated by key. If the option indicated by key takes a single
+// value, tup is unused, and the option's value is cleared regardless
+// of its current value.
 func (opts *Options) Remove(key string, tup []string) sinkdb.Op {
 	opt, ok := opts.schema[key]
 	if !ok {
 		return sinkdb.Error(errors.WithDetailf(ErrConfigOp, "Configuration option %q undefined", key))
 	}
-	if opt.tupleSize != len(tup) {
+	if opt.set && opt.tupleSize != len(tup) {
 		return sinkdb.Error(errors.WithDetailf(ErrConfigOp, "Configuration option %q expects %d arguments.", key, opt.tupleSize))
 	}
 
+	// If opt is defined to take a single value, just remove
+	// the key altogether.
+	if !opt.set {
+		return sinkdb.Delete(path.Join(sinkdbPrefix, key))
```

Comment:
Should this include an IfNotModified to protect against a concurrent update? I am not sure.